health/health.go

```diff
@@ -647,13 +665,16 @@ func (t *Tracker) updateLegacyErrorWarnableLocked(key Subsystem, err error) {
 	}
 }
 
-func (t *Tracker) SetControlHealth(problems []string) {
+func (t *Tracker) SetControlHealth(problems []tailcfg.DisplayMessage) {
```

Comment:
Need a doc here clarifying that `problems` will be sorted.

It's also not obvious to me at which point we break the loop between receiving health messages from the server and us then sending them back via [Direct.ReportHealthChange]. Is it because `w == nil`?

Reply:
Ah, I think that's what this comment is referring to: https://github.com/tailscale/tailscale/blob/1e92a12bf46e44f6358accaa19afbe55312089af/control/controlclient/direct.go#L1618-L1622

In which case I quite strongly feel that ReportHealthChange doesn't handle `w == nil` and we instead wrap the call to ReportHealthChange with a closure which discards `w == nil` with a detailed comment about why.

I think I now understand how you ended up with a second callback mechanism when we spoke about this before.

Reply:
You made me realise that we should clone `problems` because we're not supposed to mutate anything we get from `NetworkMap`.